dashboard/app/helpers/course_block_helper.rb

```diff
@@ -0,0 +1,181 @@
+module CourseBlockHelper
+  class << self
+    include Rails.application.routes.url_helpers
+    include LocaleHelper
+  end
+
+  #
+  # Given a Script ID or a string, returns an object that represents
+  # a tall course block (_course_tall_block.haml), if one exists.
+  #
+  # Example return value:
+  # {
+  #   url: "https://my.url",
+  #   title: "Tall Course Block",
+  #   body: "Click to find out more!",
+  #   teacher_guide: "https://my.teacher/guide",
+  #   image_url: "https://my.image",
+  #   audience: "Pre-reader"
+  # }
+  #
+  def self.get_tall_course_block(id)
+    default = {
+      url: script_url(id),
+      title: I18n.t("upsell.#{id}.title"),
+      body: I18n.t("upsell.#{id}.body"),
+      image_url: CDO.shared_image_url("courses/logo_tall_#{id}.jpg")
+    }
+
+    block_override_data = {
+      'applab' => {
+        url: CDO.code_org_url('/educate/applab'),
+        body: I18n.t('upsell.applab.body_short')
+      },
+      'conditionals' => {
+        url: CDO.code_org_url('/hourofcode/unplugged-conditionals-with-cards'),
+        title: I18n.t('upsell.unplugged_conditionals.title'),
+        body: I18n.t('upsell.unplugged_conditionals.body'),
+        teacher_guide: CDO.code_org_url("/hourofcode/unplugged-conditionals-with-cards")
+      },
+      'gallery' => {
+        url: '/gallery'
+      },
+      Script::APPLAB_INTRO => {
+        url: script_reset_url(Script::APPLAB_INTRO),
+        title: I18n.t('upsell.applab.title'),
+        body: I18n.t('upsell.applab.body_short')
+      },
+      Script::ARTIST_NAME => {
+        url: script_reset_url(Script::ARTIST_NAME),
+        title: data_t_suffix('script.name', id, 'title'),
+        body: data_t_suffix('script.name', id, 'description_short'),
+        teacher_guide: CDO.code_org_url("/hourofcode/artist")
+      },
+      Script::COURSE1_NAME => {
+        title: data_t_suffix('script.name', id, 'title'),
+        body: data_t_suffix('script.name', id, 'description_short'),
+        audience: data_t_suffix('script.name', id, 'description_audience')
+      },
+      Script::COURSE2_NAME => {
```

Comment:
Is there a way to factor out these duplicate sets?

Reply:
yeah, there were groups in the original HAML file that i considered keeping. my next step is going to be to standardize localizations for script title/body/etc so that this repetition gets removed altogether, so i wanted to keep everything separate for the time being

Reply:
Sounds good.  If it turns out we have a few templates for these, we could hoist them out into their own hashes, e.g. https://github.com/code-dot-org/code-dot-org/blob/staging-next/pegasus/src/homepage.rb#L354.  But removing the repetition altogether sounds better!